tsconfig.json

```diff
@@ -1,13 +1,12 @@
 {
   "compilerOptions": {
-    "baseUrl": ".",
-    "esModuleInterop": true,
-    "inlineSources": true,
-    "module": "commonjs",
-    "moduleResolution": "node",
-    "sourceMap": true,
-    "strict": true,
-    "target": "es6"
+    "noEmit": true
```

Comment:
Note that this option doesn't completely prevent files from being emitted. I noticed that when `tsc` is run with `--build` (a [new option introduced by the project references feature](https://www.typescriptlang.org/docs/handbook/project-references.html#build-mode-for-typescript)), `noEmit` is ignored, so if TypeScript is improperly configured and this file is read then files will still get emitted. That said, if `tsc` were to be run without any options then this should get honored, so I guess it is still useful.

Reply:
Hmm. That _kinda_ makes sense to me, since `--build` is for _incremental builds_. That implies something must be emitted. The concept of an incremental build relies upon there being some build to compare with, to know which parts you can skip.

Reply:
Ah, okay. Makes sense.